### PR TITLE
zephyr: cleanup src files

### PR DESCRIPTION
**Pull request by LucasTambor**
Cleanup unused or repeated src files.

### CODE REVIEW

File: zephyr/esp32c3/CMakeLists.txt [hunk 1]
```diff
@@ -193,8 +190,8 @@ if(CONFIG_SOC_SERIES_ESP32C3)
     zephyr_sources(
       ../../components/esp_hw_support/sleep_gpio.c
       ../../components/driver/gpio/rtc_io.c
-      ../../components/esp_hw_support/sleep_modes.c
-      ../../components/esp_hw_support/port/${CONFIG_SOC_SERIES}/rtc_sleep.c
+      # ../../components/esp_hw_support/sleep_modes.c
```
Comment (sylvioalves) [inline zephyr/esp32c3/CMakeLists.txt:193]: Will you remove this?

File: zephyr/esp32s2/CMakeLists.txt [hunk 2]
```diff
@@ -213,19 +211,18 @@ if(CONFIG_SOC_SERIES_ESP32S2)
...
     )
 
   if(CONFIG_PM OR CONFIG_POWEROFF)
     zephyr_sources(
       ../../components/driver/gpio/gpio.c
       ../../components/esp_hw_support/sleep_gpio.c
       ../../components/driver/gpio/rtc_io.c
-      ../../components/esp_hw_support/sleep_modes.c
-      ../../components/esp_hw_support/port/${CONFIG_SOC_SERIES}/rtc_sleep.c
+      # ../../components/esp_hw_support/sleep_modes.c
```
Comment (raffarost) [inline zephyr/esp32s2/CMakeLists.txt:222]: should this stay and instead remove L293?
although it is also used for Wi-Fi (check needed)

File: zephyr/esp32s3/CMakeLists.txt [hunk 3]
```diff
@@ -121,12 +121,10 @@ if(CONFIG_SOC_SERIES_ESP32S3)
...
   if(CONFIG_SOC_FLASH_ESP32 OR NOT CONFIG_BOOTLOADER_MCUBOOT)
     zephyr_sources(
-    ../../components/bootloader_support/bootloader_flash/src/flash_qio_mode.c
-    ../../components/bootloader_support/bootloader_flash/src/bootloader_flash.c
     ../../components/bootloader_support/src/flash_encrypt.c
     ../../components/esp_hw_support/esp_gpio_reserve.c
-    ../../components/hal/cache_hal.c
-    ../../components/hal/mmu_hal.c
+    # ../../components/hal/cache_hal.c
+    # ../../components/hal/mmu_hal.c
```
Comment (raffarost) [inline zephyr/esp32s3/CMakeLists.txt:127]: can it be removed?

File: zephyr/esp32s3/CMakeLists.txt [hunk 4]
```diff
@@ -255,12 +251,10 @@ if(CONFIG_SOC_SERIES_ESP32S3)
       ../../components/driver/gpio/gpio.c
       ../../components/esp_hw_support/sleep_gpio.c
       ../../components/driver/gpio/rtc_io.c
-      ../../components/esp_hw_support/sleep_modes.c
-      ../../components/esp_hw_support/port/${CONFIG_SOC_SERIES}/rtc_sleep.c
+      # ../../components/esp_hw_support/sleep_modes.c
```
Comment (raffarost) [inline zephyr/esp32s3/CMakeLists.txt:254]: possibly can be removed, but it's good to check wi-fi build

File: zephyr/esp32s2/CMakeLists.txt (hunk 2 above)
Comment (sylvioalves) [reply zephyr/esp32s2/CMakeLists.txt:222]: Some of it is also used in v5.1 update.

**Review by raffarost — APPROVED** [review]

**Review by wmrsouza — APPROVED** [review]

**Comment by sylvioalves** [comment]
@raffarost please revisit.

**Review by raffarost — APPROVED** [review]

**Pull request merged**